src/dispatch.rs

```diff
@@ -300,34 +311,11 @@ impl<'t> DispatcherBuilder<'t> {
     }
 }
 
-trait ExecSystem {
-    fn exec<'s>(&'s mut self, &Scope<'s>, &'s Resources, &'s AtomicBitSet);
-}
-
-struct SystemDispatch<T> {
-    id: usize,
-    system: T,
-}
-
-impl<T> SystemDispatch<T> {
-    fn new(id: usize, system: T) -> Self {
-        SystemDispatch { id: id, system: system }
-    }
-}
-
-impl<T> ExecSystem for SystemDispatch<T>
-    where T: for<'b> System<'b>
-{
-    fn exec<'s>(&'s mut self, scope: &Scope<'s>, res: &'s Resources, running: &'s AtomicBitSet) {
-        let data = T::SystemData::fetch(res);
-        scope.spawn(move |_| {
-                        self.system.work(data);
-                        running.set(self.id, false)
-                    })
-    }
-}
-
-struct SystemInfo<'t> {
+struct SystemInfo {
     dependents: Vec<usize>,
-    exec: Box<ExecSystem + Send + 't>,
 }
+
+struct SystemWorker<'a> {
+    exec: Box<FnMut(Box<Helper + 'a>, &AtomicBitSet) + Send + Sync>,
+    fetch: Box<Fn(&'a Resources) -> Box<Helper + 'a> + Send + Sync>,
```

Comment:
Adding the `Scope<'a>` into the closure makes the lifetime story quite difficult (impossible?).
That's why the fetch and exec part got split. The extra box is there to hide the return type.